RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTClusterShapeProducer.cc

```diff
@@ -88,45 +91,81 @@ void EgammaHLTClusterShapeProducer::produce(edm::StreamID sid,
   reco::RecoEcalCandidateIsolationMap clsh5x5NoiseCleanedMap(recoecalcandHandle);
 
+  //Map2 is for sigmaIPhiIPhi
+  reco::RecoEcalCandidateIsolationMap clshMap2(recoecalcandHandle);
+  reco::RecoEcalCandidateIsolationMap clsh5x5Map2(recoecalcandHandle);
+  reco::RecoEcalCandidateIsolationMap clsh5x5NoiseCleanedMap2(recoecalcandHandle);
+
   for (unsigned int iRecoEcalCand = 0; iRecoEcalCand < recoecalcandHandle->size(); iRecoEcalCand++) {
     reco::RecoEcalCandidateRef recoecalcandref(recoecalcandHandle, iRecoEcalCand);
     if (recoecalcandref->superCluster()->seed()->seed().det() != DetId::Ecal) {  //HGCAL, skip for now
       clshMap.insert(recoecalcandref, 0);
       clsh5x5Map.insert(recoecalcandref, 0);
+      clsh5x5NoiseCleanedMap.insert(recoecalcandref, 0);
+
+      clshMap2.insert(recoecalcandref, 0);
+      clsh5x5Map2.insert(recoecalcandref, 0);
+      clsh5x5NoiseCleanedMap2.insert(recoecalcandref, 0);
+
       continue;
     }
 
     double sigmaee;
-    if (EtaOrIeta_) {
+    double sigmapp;  //sigmaIphiIphi, needed in e/gamma HLT regression setup
+    if (useIEta_) {
+      //this is fractional showershape (sigmaIEtaIEta / sigmaIPhiIPhi)
       const auto& vCov = lazyTools.localCovariances(*(recoecalcandref->superCluster()->seed()));
       sigmaee = sqrt(vCov[0]);
+      sigmapp = sqrt(vCov[2]);
     } else {
+      //this is showershape using absolute geometry (sigmaEtaEta / sigmaPhiPhi)
+      //generally not used anymore
       const auto& vCov = lazyTools.covariances(*(recoecalcandref->superCluster()->seed()));
       sigmaee = sqrt(vCov[0]);
+      sigmapp = sqrt(vCov[2]);
       double EtaSC = recoecalcandref->eta();
       if (EtaSC > 1.479)
         sigmaee = sigmaee - 0.02 * (EtaSC - 2.3);
     }
 
+    //this is full5x5 showershape
     double sigmaee5x5 = sqrt(lazyTools5x5.localCovariances(*(recoecalcandref->superCluster()->seed()))[0]);
     double sigmaee5x5NoiseCleaned = sqrt(lazyTools5x5.localCovariances(*(recoecalcandref->superCluster()->seed()),
                                                                        EgammaLocalCovParamDefaults::kRelEnCut,
                                                                        &thresholds,
                                                                        multThresEB_,
                                                                        multThresEE_)[0]);
+
+    double sigmapp5x5 = sqrt(lazyTools5x5.localCovariances(*(recoecalcandref->superCluster()->seed()))[2]);
+    double sigmapp5x5NoiseCleaned = sqrt(lazyTools5x5.localCovariances(*(recoecalcandref->superCluster()->seed()),
+                                                                       EgammaLocalCovParamDefaults::kRelEnCut,
+                                                                       &thresholds,
+                                                                       multThresEB_,
+                                                                       multThresEE_)[2]);
+
```

Comment:
Looks like the 4 calls to `lazyTools5x5.localCovariances` could be reduced at least to 2. No? Something like
```cpp
    auto const ecalCandLocalCov = lazyTools5x5.localCovariances(*(recoecalcandref->superCluster()->seed()));
    auto const sigmaee5x5 = sqrt(ecalCandLocalCov[0]);
    auto const sigmapp5x5 = sqrt(ecalCandLocalCov[2]);

    auto const ecalCandLocalCovNoiseCleaned = lazyTools5x5.localCovariances(*(recoecalcandref->superCluster()->seed()), EgammaLocalCovParamDefaults::kRelEnCut, &thresholds, multThresEB_, multThresEE_);
    auto const sigmaee5x5NoiseCleaned = sqrt(ecalCandLocalCovNoiseCleaned[0]);
    auto const sigmapp5x5NoiseCleaned = sqrt(ecalCandLocalCovNoiseCleaned[2]);
```